pkg/monitoring/chain_reader.go

```diff
@@ -31,11 +32,13 @@ type chainReader struct {
 }
 
 func (c *chainReader) GetState(ctx context.Context, account solana.PublicKey, commitment rpc.CommitmentType) (state pkgSolana.State, blockHeight uint64, err error) {
-	return pkgSolana.GetState(ctx, c.client, account, commitment)
+	getReader := func() (client.AccountReader, error) { return c.client, nil }
+	return pkgSolana.GetState(ctx, getReader, account, commitment)
 }
 
 func (c *chainReader) GetLatestTransmission(ctx context.Context, account solana.PublicKey, commitment rpc.CommitmentType) (answer pkgSolana.Answer, blockHeight uint64, err error) {
-	return pkgSolana.GetLatestTransmission(ctx, c.client, account, commitment)
+	getReader := func() (client.AccountReader, error) { return c.client, nil }
+	return pkgSolana.GetLatestTransmission(ctx, getReader, account, commitment)
```

Comment:
Could you double check that `pkgSolana.GetLatestTransmission` is not used in other repos to ensure we are not breaking anything

Reply:
Yeah good call, I'll check before merging